internal/controller/monovertexrollout/monovertexrollout_controller.go

```diff
@@ -229,28 +235,50 @@ func (r *MonoVertexRolloutReconciler) reconcile(ctx context.Context, monoVertexR
 
 			monoVertexRollout.Status.MarkDeployed(monoVertexRollout.Generation)
 			r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerMonoVertexRollout, "create").Observe(time.Since(startTime).Seconds())
+
 		} else {
-			return ctrl.Result{}, fmt.Errorf("error getting MonoVertex: %v", err)
-		}
-	} else {
-		// merge and update
-		// we directly apply changes as there is no need for draining MonoVertex
-		newMonoVertexDef, err = r.merge(existingMonoVertexDef, newMonoVertexDef)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		requeueDelay, err := r.processExistingMonoVertex(ctx, monoVertexRollout, existingMonoVertexDef, newMonoVertexDef, syncStartTime)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("error processing existing MonoVertex: %v", err)
-		}
-		if requeueDelay > 0 {
-			return ctrl.Result{RequeueAfter: requeueDelay}, nil
+			existingMonoVertexDef, err := kubernetes.GetResource(ctx, r.client, newMonoVertexDef.GroupVersionKind(),
+				k8stypes.NamespacedName{Namespace: newMonoVertexDef.GetNamespace(), Name: newMonoVertexDef.GetName()})
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("error getting MonoVertex: %v", err)
+			}
+			// merge and update
+			newMonoVertexDef, err = r.merge(existingMonoVertexDef, newMonoVertexDef)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			requeueDelay, err = r.processExistingMonoVertex(ctx, monoVertexRollout, existingMonoVertexDef, newMonoVertexDef, syncStartTime)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("error processing existing MonoVertex: %v", err)
+			}
+			if requeueDelay > 0 {
+				return ctrl.Result{RequeueAfter: requeueDelay}, nil
+			}
+			// process status
+			r.processMonoVertexStatus(ctx, existingMonoVertexDef, monoVertexRollout)
```

Comment:
I ended up deciding to update this code flow, just for symmetry with the other Rollouts, although technically it wasn't necessary for this one.